### PR TITLE
geogebra: 5-0-745-0 -> 5-0-783-0, geogebra6: 6-0-745-0 -> 6-0-783-0

### PR DESCRIPTION
**Pull request by kilianar**
###### Description of changes
This pull request updates both Geogebra and Geogebra6 to their latest releases. The URL for the Geogebra package on MacOS now includes a "-x64" postfix. Since Electron version 14 is considered insecure, the Electron dependency of Geogebra6 has been updated to the latest version. I have only tested these changes on Linux x86-64, so I would appreciate it if someone could test them on Darwin.

###### Things done
- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by kilianar** [comment]
cc maintainers: @sikmir @imsofi @voidIess

**Review by soupglasses — APPROVED** [review]
While not tested extensively, i can confirm that both packages work under `x86_64-linux`. Also the change to latest electron for `geogebra6` seems to have fixed a few warnings that showed up on previous versions.

Sadly Darwin is still left untested, but i do not see anything that should create any new issues if this would be merged at least.

File: pkgs/applications/science/math/geogebra/default.nix [hunk 1]
```diff
@@ -1,7 +1,7 @@
 { lib, stdenv, fetchurl, jre, makeDesktopItem, makeWrapper, unzip, language ? "en_US" }:
 let
   pname = "geogebra";
-  version = "5-0-745-0";
+  version = "5-0-783-0";
```
Comment (sikmir) [inline pkgs/applications/science/math/geogebra/default.nix:4]: 5-0-785-0 is available.

**Comment by sikmir** [comment]
Works for me on darwin x86_64.

File: pkgs/applications/science/math/geogebra/geogebra6.nix [hunk 2]
```diff
@@ -1,7 +1,7 @@
 { lib, stdenv, unzip, fetchurl, electron, makeWrapper, geogebra }:
 let
   pname = "geogebra";
-  version = "6-0-745-0";
+  version = "6-0-783-0";
```
Comment (sikmir) [inline pkgs/applications/science/math/geogebra/geogebra6.nix:4]:
```suggestion
  version = "6-0-785-0";
```

File: pkgs/applications/science/math/geogebra/default.nix (hunk 1 above)
Comment (kilianar) [reply pkgs/applications/science/math/geogebra/default.nix]: Thanks, I updated the pull request accordingly.

**Review by marsam — APPROVED** [review]

**Pull request merged**